# Capstone Project - The Batttle of the Neighborhoods Final

## IBM Data Science Professional Certificate Applied Capstone Project

## Business Problem

Clearly define a problem or an idea of your choice, where you would need to leverage the Foursquare location data to solve or execute. Remember that data science problems always target an audience and are meant to help a group of stakeholders solve a problem, so make sure that you explicitly describe your audience and why they would care about your problem.

#### The problem that is to be solved by our application is where is the best neighbhorhood in Toronto for our clients or stakeholders to move into based on what amenties are located nearby.

#### Toronto is considered one of the most popular cities in Canada. Many people make the decision to move there every year to different neighbhorhoods for a variety of reasons. A number of the existing popluation even choose to move to different places within the city for various reasons. By utilizing FourSquare location data, we will be able to segment and cluster neighborhoods together and collect data on what amenties are in each neighborhood. By collecting type and frequency of venues nearby, the client or stakeholder will be able to determine where they think is the best place for them to move. As apart of a larger question to be solved, many choose their residence based on what is nearby making this application useful to those looking to move to a neighborhood close to their favorite businesses or places.

## Data

Describe the data that you will be using to solve the problem or execute your idea. Remember that you will need to use the Foursquare location data to solve the problem or execute your idea. You can absolutely use other datasets in combination with the Foursquare location data. So make sure that you provide adequate explanation and discussion, with examples, of the data that you will be using, even if it is only Foursquare location data.

#### In order to provide necessary information to our clients or to the stakeholders, Foursquare API to collect data such as type and frequency for venues in each neighborhood. By segmenting and clustering, we can find the top and most popular venues in each neighborhood in Toronto. This will give the end user an idea of what kind of ammenties are and where there are located.

#### Data to be used in this project will be as follows:

#### List of postal codes of Canada Wikipedia for neighborhoods: https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M

#### Foursquare API data for amenties, shops, and resturants in each neighborhood of Toronto

### Import Libraries

In [1]:
import numpy as np
import pandas as pd 
import json

!pip install wikipedia
import wikipedia as wp

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

!conda install -c conda-forge geopy --yes

from geopy.geocoders import Nominatim 
import requests 
from pandas.io.json import json_normalize 

import matplotlib.cm as cm
import matplotlib.colors as colors

from sklearn.cluster import KMeans

!conda install -c conda-forge folium=0.5.0 --yes
import folium

print("Import Complete")

  Stored in directory: /home/dsxuser/.cache/pip/wheels/87/2a/18/4e471fd96d12114d16fe4a446d00c3b38fb9efcb744bd31f4a
Successfully built wikipedia
Solving environment: done

## Package Plan ##

  environment location: /opt/conda/envs/Python36

  added / updated specs: 
    - geopy


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2020.6.20  |       hecda079_0         145 KB  conda-forge
    python_abi-3.6             |          1_cp36m           4 KB  conda-forge
    geopy-2.0.0                |     pyh9f0ad1d_0          63 KB  conda-forge
    certifi-2020.6.20          |   py36h9f0ad1d_0         151 KB  conda-forge
    openssl-1.1.1g             |       h516909a_1         2.1 MB  conda-forge
    geographiclib-1.50         |             py_0          34 KB  conda-forge
    ------------------------------------------------------------
                                          

### Create Table of Toronto Neighborhoods with Geographical coordinates.

We need to know where exactly each neighbhorhood is located in Toronto. We first create a table containing neighborhood name and it's geographical location using longitude and latitude.

In [2]:
html = wp.page("List of postal codes of Canada: M").html().encode("UTF-8")

df = pd.read_html(html)[0]
df = df[df.Community!= "Not assigned"]

df = df.groupby(['Postal Code', 'Community'])['Neighbourhood'].apply(list).apply(lambda x:', '.join(x)).to_frame().reset_index()
for index, row in df.iterrows():
    if row['Neighbourhood'] == 'Not assigned':
        row['Neighbourhood'] = row['Community']

column_names = ['Postal Code','Community', 'Neighborhood'] 
df.columns = column_names

df.head()

Postal Code    Community                            Neighborhood
0         M1B  Scarborough                          Malvern, Rouge
1         M1C  Scarborough  Rouge Hill, Port Union, Highland Creek
2         M1E  Scarborough       Guildwood, Morningside, West Hill
3         M1G  Scarborough                                  Woburn
4         M1H  Scarborough                               Cedarbrae

In [3]:
import pandas as pd
import io
import requests

url="https://cocl.us/Geospatial_data"

s=requests.get(url).content

c=pd.read_csv(io.StringIO(s.decode('utf-8')))

dfc = df.join(c.set_index('Postal Code'), on='Postal Code')

dfc.head()

Postal Code    Community                            Neighborhood   Latitude  \
0         M1B  Scarborough                          Malvern, Rouge  43.806686   
1         M1C  Scarborough  Rouge Hill, Port Union, Highland Creek  43.784535   
2         M1E  Scarborough       Guildwood, Morningside, West Hill  43.763573   
3         M1G  Scarborough                                  Woburn  43.770992   
4         M1H  Scarborough                               Cedarbrae  43.773136   

   Longitude  
0 -79.194353  
1 -79.160497  
2 -79.188711  
3 -79.216917  
4 -79.239476

In [4]:
address = 'Toronto'
geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude

print('Coordinates of Toronto are {}, {}.'.format(latitude, longitude))

Coordinates of Toronto are 43.6534817, -79.3839347.


### Create Map of Toronto

To get a more visual understanding or where the neighborhoods are located in Toronto, we create a map using the Folium Map package.

In [5]:
map_toronto = folium.Map(location=[latitude, longitude], zoom_start=10)

for lat, lng, district, neighborhood in zip(dfc['Latitude'], dfc['Longitude'], dfc['Community'], dfc['Neighborhood']):
    label = '{}, {}'.format(neighborhood, district)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=4,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.2,
        parse_html=False).add_to(map_toronto)  
    
map_toronto

### FourSquare API

By using FourSquare API, we will be able to search and collect popular venues in Toronto. We will see what type, location, and frequency of specific venues located in the different neighborhoods. By using location data, we will be able to determine which neighborhood best suits the client or stakeholders needs and wants.

In [6]:
CLIENT_ID = '0LEACYSZ1UDN2OT3EADC5KUX4KYXBCUBVR0CC44RIB0SAD0N' 
CLIENT_SECRET = 'HTYT3TIWRTL2YAZWEVJVFFS4RT3RTSKBPIIOK0AWVN2RAQV3'

VERSION = '20180604'
LIMIT = 100

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)
radius = 500

url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude, VERSION, radius, LIMIT)
url

Your credentails:
CLIENT_ID: 0LEACYSZ1UDN2OT3EADC5KUX4KYXBCUBVR0CC44RIB0SAD0N
CLIENT_SECRET:HTYT3TIWRTL2YAZWEVJVFFS4RT3RTSKBPIIOK0AWVN2RAQV3


'https://api.foursquare.com/v2/venues/search?client_id=0LEACYSZ1UDN2OT3EADC5KUX4KYXBCUBVR0CC44RIB0SAD0N&client_secret=HTYT3TIWRTL2YAZWEVJVFFS4RT3RTSKBPIIOK0AWVN2RAQV3&ll=43.6534817,-79.3839347&v=20180604&radius=500&limit=100'

In [7]:
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

In [8]:
results = requests.get(url).json()
results

venues = results['response']['venues']

dataframe = json_normalize(venues)
dataframe.head()

categories  hasPerk  \
0  [{'id': '4bf58dd8d48988d129941735', 'name': 'C...    False   
1  [{'id': '4bf58dd8d48988d129941735', 'name': 'C...    False   
2  [{'id': '4bf58dd8d48988d176941735', 'name': 'G...    False   
3  [{'id': '52e81612bcbc57f1066b7a0c', 'name': 'B...    False   
4  [{'id': '4bf58dd8d48988d129941735', 'name': 'C...    False   

                         id   location.address location.cc location.city  \
0  4c093ee0340720a153728493                NaN          CA           NaN   
1  4ad4c05ef964a5208ff620e3   100 Queen St. W.          CA       Toronto   
2  50885719498ea7b5aab3a74c         483 Bay St          CA       Toronto   
3  5f5e5ad065f5f93e83216382  258 Queens Quay W          CA       Toronto   
4  5b193c42598e64002ca79b96     100 Queen St W          CA       Toronto   

  location.country location.crossStreet  location.distance  \
0           Canada                  NaN                184   
1           Canada           at Bay St.                 38   
2           Canada                  NaN                130   
3           Canada                  NaN               1573   
4           Canada                  NaN                  3   

                           location.formattedAddress  \
0                                           [Canada]   
1  [100 Queen St. W. (at Bay St.), Toronto ON M5H...   
2           [483 Bay St, Toronto ON M5G 2C9, Canada]   
3    [258 Queens Quay W, Toronto ON M5J 1B5, Canada]   
4       [100 Queen St W, Toronto ON M5H 2N2, Canada]   

                             location.labeledLatLngs  location.lat  \
0  [{'label': 'display', 'lat': 43.65182710471462...     43.651827   
1  [{'label': 'display', 'lat': 43.65313989695342...     43.653140   
2  [{'label': 'display', 'lat': 43.653436, 'lng':...     43.653436   
3  [{'label': 'display', 'lat': 43.639397, 'lng':...     43.639397   
4  [{'label': 'display', 'lat': 43.653454, 'lng':...     43.653454   

   location.lng location.neighborhood location.postalCode location.state  \
0    -79.383949                   NaN                 NaN            NaN   
1    -79.383967                   NaN             M5H 2N2             ON   
2    -79.382314                   NaN             M5G 2C9             ON   
3    -79.385605                   NaN             M5J 1B5             ON   
4    -79.383952                   NaN             M5H 2N2             ON   

                                           name    referralId  
0                    City Hall Council Chambers  v-1600556191  
1                             Toronto City Hall  v-1600556191  
2  GoodLife Fitness Toronto Bell Trinity Centre  v-1600556191  
3                                   Bubble Baby  v-1600556191  
4       City of Toronto Civic Innovation Office  v-1600556191

In [9]:
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']

dataframe_filtered = dataframe.loc[:, filtered_columns]

def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]
dataframe_filtered.head()

name       categories  \
0                    City Hall Council Chambers        City Hall   
1                             Toronto City Hall        City Hall   
2  GoodLife Fitness Toronto Bell Trinity Centre              Gym   
3                                   Bubble Baby  Bubble Tea Shop   
4       City of Toronto Civic Innovation Office        City Hall   

             address  cc     city country crossStreet  distance  \
0                NaN  CA      NaN  Canada         NaN       184   
1   100 Queen St. W.  CA  Toronto  Canada  at Bay St.        38   
2         483 Bay St  CA  Toronto  Canada         NaN       130   
3  258 Queens Quay W  CA  Toronto  Canada         NaN      1573   
4     100 Queen St W  CA  Toronto  Canada         NaN         3   

                                    formattedAddress  \
0                                           [Canada]   
1  [100 Queen St. W. (at Bay St.), Toronto ON M5H...   
2           [483 Bay St, Toronto ON M5G 2C9, Canada]   
3    [258 Queens Quay W, Toronto ON M5J 1B5, Canada]   
4       [100 Queen St W, Toronto ON M5H 2N2, Canada]   

                                      labeledLatLngs        lat        lng  \
0  [{'label': 'display', 'lat': 43.65182710471462...  43.651827 -79.383949   
1  [{'label': 'display', 'lat': 43.65313989695342...  43.653140 -79.383967   
2  [{'label': 'display', 'lat': 43.653436, 'lng':...  43.653436 -79.382314   
3  [{'label': 'display', 'lat': 43.639397, 'lng':...  43.639397 -79.385605   
4  [{'label': 'display', 'lat': 43.653454, 'lng':...  43.653454 -79.383952   

  neighborhood postalCode state                        id  
0          NaN        NaN   NaN  4c093ee0340720a153728493  
1          NaN    M5H 2N2    ON  4ad4c05ef964a5208ff620e3  
2          NaN    M5G 2C9    ON  50885719498ea7b5aab3a74c  
3          NaN    M5J 1B5    ON  5f5e5ad065f5f93e83216382  
4          NaN    M5H 2N2    ON  5b193c42598e64002ca79b96

In [10]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [11]:
toronto_venues = getNearbyVenues(names=dfc['Neighborhood'],
                                   latitudes=dfc['Latitude'],
                                   longitudes=dfc['Longitude']
                                  )

toronto_venues.head()

Malvern, Rouge
Rouge Hill, Port Union, Highland Creek
Guildwood, Morningside, West Hill
Woburn
Cedarbrae
Scarborough Village
Kennedy Park, Ionview, East Birchmount Park
Golden Mile, Clairlea, Oakridge
Cliffside, Cliffcrest, Scarborough Village West
Birch Cliff, Cliffside West
Dorset Park, Wexford Heights, Scarborough Town Centre
Wexford, Maryvale
Agincourt
Clarks Corners, Tam O'Shanter, Sullivan
Milliken, Agincourt North, Steeles East, L'Amoreaux East
Steeles West, L'Amoreaux West
Upper Rouge
Hillcrest Village
Fairview, Henry Farm, Oriole
Bayview Village
York Mills, Silver Hills
Willowdale, Newtonbrook
Willowdale, Willowdale East
York Mills West
Willowdale, Willowdale West
Parkwoods
Don Mills
Don Mills
Bathurst Manor, Wilson Heights, Downsview North
Northwood Park, York University
Downsview
Downsview
Downsview
Downsview
Victoria Village
Parkview Hill, Woodbine Gardens
Woodbine Heights
The Beaches
Leaside
Thorncliffe Park
East Toronto, Broadview North (Old East York)
The Danforth West, 

Neighborhood  Neighborhood Latitude  \
0                          Malvern, Rouge              43.806686   
1  Rouge Hill, Port Union, Highland Creek              43.784535   
2  Rouge Hill, Port Union, Highland Creek              43.784535   
3       Guildwood, Morningside, West Hill              43.763573   
4       Guildwood, Morningside, West Hill              43.763573   

   Neighborhood Longitude                                              Venue  \
0              -79.194353                                            Wendy’s   
1              -79.160497                              Royal Canadian Legion   
2              -79.160497  SEBS Engineering Inc. (Sustainable Energy and ...   
3              -79.188711                                     RBC Royal Bank   
4              -79.188711                                  G & G Electronics   

   Venue Latitude  Venue Longitude              Venue Category  
0       43.807448       -79.199056        Fast Food Restaurant  
1       43.782533       -79.163085                         Bar  
2       43.782371       -79.156820  Construction & Landscaping  
3       43.766790       -79.191151                        Bank  
4       43.765309       -79.191537           Electronics Store

In [12]:
print('{} venues were returned by Foursquare.'.format(toronto_venues.shape[0]))
toronto_venues.head()

2145 venues were returned by Foursquare.


Neighborhood  Neighborhood Latitude  \
0                          Malvern, Rouge              43.806686   
1  Rouge Hill, Port Union, Highland Creek              43.784535   
2  Rouge Hill, Port Union, Highland Creek              43.784535   
3       Guildwood, Morningside, West Hill              43.763573   
4       Guildwood, Morningside, West Hill              43.763573   

   Neighborhood Longitude                                              Venue  \
0              -79.194353                                            Wendy’s   
1              -79.160497                              Royal Canadian Legion   
2              -79.160497  SEBS Engineering Inc. (Sustainable Energy and ...   
3              -79.188711                                     RBC Royal Bank   
4              -79.188711                                  G & G Electronics   

   Venue Latitude  Venue Longitude              Venue Category  
0       43.807448       -79.199056        Fast Food Restaurant  
1       43.782533       -79.163085                         Bar  
2       43.782371       -79.156820  Construction & Landscaping  
3       43.766790       -79.191151                        Bank  
4       43.765309       -79.191537           Electronics Store

In [13]:
toronto_venues.groupby('Neighborhood').count()

Neighborhood Latitude  \
Neighborhood                                                                
Agincourt                                                               5   
Alderwood, Long Branch                                                  7   
Bathurst Manor, Wilson Heights, Downsview North                        21   
Bayview Village                                                         4   
Bedford Park, Lawrence Manor East                                      22   
Berczy Park                                                            55   
Birch Cliff, Cliffside West                                             4   
Brockton, Parkdale Village, Exhibition Place                           23   
Business reply mail Processing Centre, South Ce...                     16   
CN Tower, King and Spadina, Railway Lands, Harb...                     16   
Caledonia-Fairbanks                                                     4   
Canada Post Gateway Processing Centre                                  13   
Cedarbrae                                                               8   
Central Bay Street                                                     71   
Christie                                                               16   
Church and Wellesley                                                   77   
Clarks Corners, Tam O'Shanter, Sullivan                                12   
Cliffside, Cliffcrest, Scarborough Village West                         2   
Commerce Court, Victoria Hotel                                        100   
Davisville                                                             33   
Davisville North                                                        9   
Del Ray, Mount Dennis, Keelsdale and Silverthorn                        5   
Don Mills                                                              23   
Dorset Park, Wexford Heights, Scarborough Town ...                      5   
Downsview                                                              16   
Dufferin, Dovercourt Village                                           14   
East Toronto, Broadview North (Old East York)                           3   
Eringate, Bloordale Gardens, Old Burnhamthorpe,...                      8   
Fairview, Henry Farm, Oriole                                           70   
First Canadian Place, Underground city                                100   
Forest Hill North & West, Forest Hill Road Park                         4   
Garden District, Ryerson                                              100   
Glencairn                                                               4   
Golden Mile, Clairlea, Oakridge                                        10   
Guildwood, Morningside, West Hill                                       8   
Harbourfront East, Union Station, Toronto Islands                     100   
High Park, The Junction South                                          25   
Hillcrest Village                                                       5   
Humber Summit                                                           2   
Humberlea, Emery                                                        1   
Humewood-Cedarvale                                                      4   
India Bazaar, The Beaches West                                         19   
Kennedy Park, Ionview, East Birchmount Park                             4   
Kensington Market, Chinatown, Grange Park                              74   
Kingsview Village, St. Phillips, Martin Grove G...                      4   
Lawrence Manor, Lawrence Heights                                       13   
Lawrence Park                                                           3   
Leaside                                                                33   
Little Portugal, Trinity                                               45   
Malvern, Rouge                                                          1   
Milliken, Agincourt North, Steeles East, L'Amor.

In [14]:
print('There are {} uniques categories.'.format(len(toronto_venues['Venue Category'].unique())))

There are 274 uniques categories.


This is where we use Foursquare API to start to look at what type of venues are there, frequency, and location in the neighborhoods. The information given by FourSquare API will give our clients or stakeholders the exact information they are looking for in order to make their decision on which neighborhood to move too.

In [15]:
#### One Hot Encoding

toronto_onehot = pd.get_dummies(toronto_venues[['Venue Category']], prefix="", prefix_sep="")
toronto_onehot['Neighborhood'] = toronto_venues['Neighborhood'] 

fixed_columns = [toronto_onehot.columns[-1]] + list(toronto_onehot.columns[:-1])

manhattan_onehot = toronto_onehot[fixed_columns]

toronto_onehot.head()

Accessories Store  Afghan Restaurant  Airport  Airport Food Court  \
0                  0                  0        0                   0   
1                  0                  0        0                   0   
2                  0                  0        0                   0   
3                  0                  0        0                   0   
4                  0                  0        0                   0   

   Airport Gate  Airport Lounge  Airport Service  Airport Terminal  \
0             0               0                0                 0   
1             0               0                0                 0   
2             0               0                0                 0   
3             0               0                0                 0   
4             0               0                0                 0   

   American Restaurant  Antique Shop  Aquarium  Art Gallery  Art Museum  \
0                    0             0         0            0           0   
1                    0             0         0            0           0   
2                    0             0         0            0           0   
3                    0             0         0            0           0   
4                    0             0         0            0           0   

   Arts & Crafts Store  Asian Restaurant  Athletics & Sports  Auto Garage  \
0                    0                 0                   0            0   
1                    0                 0                   0            0   
2                    0                 0                   0            0   
3                    0                 0                   0            0   
4                    0                 0                   0            0   

   Auto Workshop  BBQ Joint  Baby Store  Bagel Shop  Bakery  Bank  Bar  \
0              0          0           0           0       0     0    0   
1              0          0           0           0       0     0    1   
2              0          0           0           0       0     0    0   
3              0          0           0           0       0     1    0   
4              0          0           0           0       0     0    0   

   Baseball Field  Baseball Stadium  Basketball Stadium  Beach  \
0               0                 0                   0      0   
1               0                 0                   0      0   
2               0                 0                   0      0   
3               0                 0                   0      0   
4               0                 0                   0      0   

   Bed & Breakfast  Beer Bar  Beer Store  Belgian Restaurant  Bike Shop  \
0                0         0           0                   0          0   
1                0         0           0                   0          0   
2                0         0           0                   0          0   
3                0         0           0                   0          0   
4                0         0           0                   0          0   

   Bistro  Boat or Ferry  Bookstore  Boutique  Brazilian Restaurant  \
0       0              0          0         0                     0   
1       0              0          0         0                     0   
2       0              0          0         0                     0   
3       0              0          0         0                     0   
4       0              0          0         0                     0   

   Breakfast Spot  Brewery  Bridal Shop  Bubble Tea Shop  Building  \
0               0        0            0                0         0   
1               0        0            0                0         0   
2               0        0            0                0         0   
3               0        0            0                0         0   
4               0        0            0                0         0   

   Burger Joint  Burrito Place  Bus Line  Bus Station  Business Service  \
0             0    

In [16]:
toronto_grouped = toronto_onehot.groupby('Neighborhood').mean().reset_index()

toronto_grouped.head()

Neighborhood  Accessories Store  \
0                                        Agincourt                0.0   
1                           Alderwood, Long Branch                0.0   
2  Bathurst Manor, Wilson Heights, Downsview North                0.0   
3                                  Bayview Village                0.0   
4                Bedford Park, Lawrence Manor East                0.0   

   Afghan Restaurant  Airport  Airport Food Court  Airport Gate  \
0                0.0      0.0                 0.0           0.0   
1                0.0      0.0                 0.0           0.0   
2                0.0      0.0                 0.0           0.0   
3                0.0      0.0                 0.0           0.0   
4                0.0      0.0                 0.0           0.0   

   Airport Lounge  Airport Service  Airport Terminal  American Restaurant  \
0             0.0              0.0               0.0             0.000000   
1             0.0              0.0               0.0             0.000000   
2             0.0              0.0               0.0             0.000000   
3             0.0              0.0               0.0             0.000000   
4             0.0              0.0               0.0             0.045455   

   Antique Shop  Aquarium  Art Gallery  Art Museum  Arts & Crafts Store  \
0           0.0       0.0          0.0         0.0                  0.0   
1           0.0       0.0          0.0         0.0                  0.0   
2           0.0       0.0          0.0         0.0                  0.0   
3           0.0       0.0          0.0         0.0                  0.0   
4           0.0       0.0          0.0         0.0                  0.0   

   Asian Restaurant  Athletics & Sports  Auto Garage  Auto Workshop  \
0               0.0                 0.0          0.0            0.0   
1               0.0                 0.0          0.0            0.0   
2               0.0                 0.0          0.0            0.0   
3               0.0                 0.0          0.0            0.0   
4               0.0                 0.0          0.0            0.0   

   BBQ Joint  Baby Store  Bagel Shop  Bakery      Bank  Bar  Baseball Field  \
0        0.0         0.0         0.0     0.0  0.000000  0.0             0.0   
1        0.0         0.0         0.0     0.0  0.000000  0.0             0.0   
2        0.0         0.0         0.0     0.0  0.095238  0.0             0.0   
3        0.0         0.0         0.0     0.0  0.250000  0.0             0.0   
4        0.0         0.0         0.0     0.0  0.000000  0.0             0.0   

   Baseball Stadium  Basketball Stadium  Beach  Bed & Breakfast  Beer Bar  \
0               0.0                 0.0    0.0              0.0       0.0   
1               0.0                 0.0    0.0              0.0       0.0   
2               0.0                 0.0    0.0              0.0       0.0   
3               0.0                 0.0    0.0              0.0       0.0   
4               0.0                 0.0    0.0              0.0       0.0   

   Beer Store  Belgian Restaurant  Bike Shop  Bistro  Boat or Ferry  \
0         0.0                 0.0        0.0     0.0            0.0   
1         0.0                 0.0        0.0     0.0            0.0   
2         0.0                 0.0        0.0     0.0            0.0   
3         0.0                 0.0        0.0     0.0            0.0   
4         0.0                 0.0        0.0     0.0            0.0   

   Bookstore  Boutique  Brazilian Restaurant  Breakfast Spot  Brewery  \
0        0.0       0.0                   0.0             0.2      0.0   
1        0.0       0.0                   0.0             0.0      0.0   
2        0.0       0.0                   0.0             0.0      0.0   
3        0.0       0.0                   0.0             0.0      0.0   
4        0.0       0.0                   0.0             0.0      0.0   

   Bridal Shop  Bubble Tea Shop  Building  Burger J

We will use the top 10 most popular venues in order to get a better picture of what each of the neighborhoods are like. By using 10 different venues we will see various resturants, stores, and other businesses in the area that people might want to visit regularly. 

In [17]:
num_top_venues = 11

for hood in toronto_grouped['Neighborhood']:
    
    print("----"+hood+"----")
    
    temp = toronto_grouped[toronto_grouped['Neighborhood'] == hood].T.reset_index()
    temp.columns = ['venue','freq']
    temp = temp.iloc[1:]
    temp['freq'] = temp['freq'].astype(float)
    temp = temp.round({'freq': 2})
   
    print(temp.sort_values('freq', ascending=False).reset_index(drop=True).head(num_top_venues))
    print('\n')

----Agincourt----
                              venue  freq
0         Latin American Restaurant   0.2
1                    Breakfast Spot   0.2
2                            Lounge   0.2
3                      Skating Rink   0.2
4                    Clothing Store   0.2
5                 Accessories Store   0.0
6               Monument / Landmark   0.0
7   Molecular Gastronomy Restaurant   0.0
8        Modern European Restaurant   0.0
9                 Mobile Phone Shop   0.0
10               Miscellaneous Shop   0.0


----Alderwood, Long Branch----
                       venue  freq
0                Pizza Place  0.29
1                        Gym  0.14
2                        Pub  0.14
3                Coffee Shop  0.14
4                   Pharmacy  0.14
5             Sandwich Place  0.14
6   Mediterranean Restaurant  0.00
7                Men's Store  0.00
8             Medical Center  0.00
9              Metro Station  0.00
10        Mexican Restaurant  0.00


----Bathurst Manor, Wil

                   venue  freq
0         Breakfast Spot  0.11
1                  Hotel  0.11
2       Department Store  0.11
3           Dance Studio  0.11
4         Sandwich Place  0.11
5   Gym / Fitness Center  0.11
6                Dog Run  0.11
7      Food & Drink Shop  0.11
8                   Park  0.11
9          Metro Station  0.00
10    Mexican Restaurant  0.00


----Del Ray, Mount Dennis, Keelsdale and Silverthorn----
                        venue  freq
0              Discount Store   0.2
1                  Restaurant   0.2
2                         Bar   0.2
3          Turkish Restaurant   0.2
4              Sandwich Place   0.2
5    Mediterranean Restaurant   0.0
6                 Men's Store   0.0
7               Metro Station   0.0
8          Mexican Restaurant   0.0
9   Middle Eastern Restaurant   0.0
10                      Motel   0.0


----Don Mills----
                   venue  freq
0                    Gym  0.13
1             Beer Store  0.09
2            Coffee Shop

                              venue  freq
0                       Pizza Place   0.5
1            Furniture / Home Store   0.5
2         Middle Eastern Restaurant   0.0
3               Monument / Landmark   0.0
4   Molecular Gastronomy Restaurant   0.0
5        Modern European Restaurant   0.0
6                 Mobile Phone Shop   0.0
7                Miscellaneous Shop   0.0
8                Mexican Restaurant   0.0
9                  Malay Restaurant   0.0
10                    Metro Station   0.0


----Humberlea, Emery----
                              venue  freq
0                    Baseball Field   1.0
1                 Accessories Store   0.0
2         Middle Eastern Restaurant   0.0
3               Monument / Landmark   0.0
4   Molecular Gastronomy Restaurant   0.0
5        Modern European Restaurant   0.0
6                 Mobile Phone Shop   0.0
7                Miscellaneous Shop   0.0
8                Mexican Restaurant   0.0
9                             Motel   0.0
10     

                        venue  freq
0              Massage Studio  0.14
1                 Coffee Shop  0.14
2        Caribbean Restaurant  0.14
3                         Bar  0.14
4      Furniture / Home Store  0.14
5          Miscellaneous Shop  0.14
6               Metro Station  0.14
7          Mexican Restaurant  0.00
8   Middle Eastern Restaurant  0.00
9               Movie Theater  0.00
10          Mobile Phone Shop  0.00


----Old Mill South, King's Mill Park, Sunnylea, Humber Bay, Mimico NE, The Queensway East, Royal York South East, Kingsway Park South East----
                              venue  freq
0        Construction & Landscaping   0.5
1                    Baseball Field   0.5
2                 Accessories Store   0.0
3                Miscellaneous Shop   0.0
4               Moroccan Restaurant   0.0
5               Monument / Landmark   0.0
6   Molecular Gastronomy Restaurant   0.0
7        Modern European Restaurant   0.0
8                 Mobile Phone Shop   0.0
9  

                  venue  freq
0           Coffee Shop  0.13
1           Pizza Place  0.07
2            Sports Bar  0.07
3                  Bank  0.07
4            Bagel Shop  0.07
5           Supermarket  0.07
6            Restaurant  0.07
7    Light Rail Station  0.07
8   Fried Chicken Joint  0.07
9                   Pub  0.07
10  American Restaurant  0.07


----The Annex, North Midtown, Yorkville----
                        venue  freq
0              Sandwich Place  0.16
1                        Café  0.16
2                 Coffee Shop  0.11
3                         Pub  0.05
4                  Donut Shop  0.05
5                   BBQ Joint  0.05
6                Liquor Store  0.05
7                Burger Joint  0.05
8                 Pizza Place  0.05
9   Middle Eastern Restaurant  0.05
10                       Park  0.05


----The Beaches----
                              venue  freq
0                 Health Food Store  0.25
1                             Trail  0.25
2             

                              venue  freq
0                 Convenience Store   0.5
1                              Park   0.5
2                 Accessories Store   0.0
3                Mexican Restaurant   0.0
4               Monument / Landmark   0.0
5   Molecular Gastronomy Restaurant   0.0
6        Modern European Restaurant   0.0
7                 Mobile Phone Shop   0.0
8                Miscellaneous Shop   0.0
9         Middle Eastern Restaurant   0.0
10                    Metro Station   0.0


----York Mills, Silver Hills----
                              venue  freq
0               Martial Arts School   1.0
1                 Accessories Store   0.0
2         Middle Eastern Restaurant   0.0
3               Moroccan Restaurant   0.0
4               Monument / Landmark   0.0
5   Molecular Gastronomy Restaurant   0.0
6        Modern European Restaurant   0.0
7                 Mobile Phone Shop   0.0
8                Miscellaneous Shop   0.0
9                Mexican Restaurant   0.0

In [18]:
def return_most_common_venues(row, num_top_venues):
    
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

In [19]:
num_top_venues = 10
indicators = ['st', 'nd', 'rd']
columns = ['Neighborhood']

for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighborhood'] = toronto_grouped['Neighborhood']

for ind in np.arange(toronto_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(toronto_grouped.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted.head()

Neighborhood 1st Most Common Venue  \
0                                        Agincourt                Lounge   
1                           Alderwood, Long Branch           Pizza Place   
2  Bathurst Manor, Wilson Heights, Downsview North                  Bank   
3                                  Bayview Village                  Café   
4                Bedford Park, Lawrence Manor East        Sandwich Place   

  2nd Most Common Venue      3rd Most Common Venue 4th Most Common Venue  \
0        Breakfast Spot  Latin American Restaurant          Skating Rink   
1        Sandwich Place                Coffee Shop                   Pub   
2           Coffee Shop                   Pharmacy            Restaurant   
3   Japanese Restaurant                       Bank    Chinese Restaurant   
4    Italian Restaurant                Coffee Shop         Grocery Store   

  5th Most Common Venue 6th Most Common Venue    7th Most Common Venue  \
0        Clothing Store   Dumpling Restaurant      Distribution Center   
1              Pharmacy                   Gym             Gourmet Shop   
2           Supermarket         Deli / Bodega         Sushi Restaurant   
3           Yoga Studio   Distribution Center                  Dog Run   
4       Thai Restaurant          Liquor Store  Comfort Food Restaurant   

  8th Most Common Venue 9th Most Common Venue 10th Most Common Venue  
0               Dog Run      Doner Restaurant             Donut Shop  
1        Discount Store         Deli / Bodega       Department Store  
2     Mobile Phone Shop   Fried Chicken Joint         Ice Cream Shop  
3      Doner Restaurant            Donut Shop              Drugstore  
4             Juice Bar               Butcher                   Café

By this point, we are starting to see the make up of neighborhoods around Toronto. As you can see in the table, FourSquare API can give very specific information about the businesses that can be found. By using that information we are able to cater our application to our clients and stakeholders. While many people may put the same importance on the first few venues, most common venues 4 - 10 can greatly differ in importance to different users. The more important venues show up on the list, the more likely a user is to find that neighborhood more attractive and willing to move there. 

## Methodology

In this project, we took a look at what type of venues, where they are located, and how often does a specific type of venue appear within the various neighborhoods in Toronto. 

First, our application took neighborhood data from a wikipedia page in order to find the exact location (longitude and latitude) of each of the different neighborhoods. After marking the locations of each neighborhood, we were able to visualize where the neighborhoods are located by producing a map from the folium package. 

Second, we used FourSquare API location data to find all the different types of venues located in the various neighborhoods around Toronto. This gave us an idea of what type and how many venues are located within a specific neighborhood. FourSquare returned **2145** venues around Toronto. This produced **274** unique categories that we can use to organize our data in. 

Third, we were able to take the FourSquare and location data and place them into the neighborhoods they are located in. From there we can cluster the neighborhoods together then create a list of the **10** most common venues found.

Finally, after looking through the list of neighborhoods and their venues, we can choose which neighborhoods hold the most interest for our users and compare them. By comparing the neighborhoods side by side, we can see their locations and what the **10** most common venues are next to each other. This will give our user a very visual display of what the neighborhoods are like and use that information to help them make a decision.

## Analysis

Next we will cluster the neighborhoods together so we can get a better picture of what venues are where and even which neighborhoods are close to each other. This will allow our user to analize the data and see not only what is in their neighborhood of choice, but also what may be nearby for them to travel too. After clustering the neighborhoods, we will be able to compare neighborhoods together to better see which neighborhoods suit our users best.

### Cluster Neighborhoods 

In [20]:
kclusters = 2
toronto_grouped_clustering = toronto_grouped.drop('Neighborhood', 1)

kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(toronto_grouped_clustering)
kmeans.labels_[0:10]

array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1], dtype=int32)

In [21]:
neighborhoods_venues_sorted.insert(0, 'ClusterLabels', kmeans.labels_)
toronto_merged = dfc

toronto_merged = toronto_merged.join(neighborhoods_venues_sorted.set_index('Neighborhood'), on='Neighborhood')
toronto_merged = toronto_merged.dropna()
toronto_merged.head()

Postal Code    Community                            Neighborhood   Latitude  \
0         M1B  Scarborough                          Malvern, Rouge  43.806686   
1         M1C  Scarborough  Rouge Hill, Port Union, Highland Creek  43.784535   
2         M1E  Scarborough       Guildwood, Morningside, West Hill  43.763573   
3         M1G  Scarborough                                  Woburn  43.770992   
4         M1H  Scarborough                               Cedarbrae  43.773136   

   Longitude  ClusterLabels       1st Most Common Venue 2nd Most Common Venue  \
0 -79.194353            1.0        Fast Food Restaurant             Drugstore   
1 -79.160497            1.0  Construction & Landscaping                   Bar   
2 -79.188711            1.0                Intersection    Mexican Restaurant   
3 -79.216917            1.0                 Coffee Shop     Korean Restaurant   
4 -79.239476            1.0        Caribbean Restaurant   Fried Chicken Joint   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0                 Diner        Discount Store   Distribution Center   
1           Yoga Studio   Dumpling Restaurant   Distribution Center   
2        Breakfast Spot     Electronics Store            Restaurant   
3    Mexican Restaurant             Drugstore        Discount Store   
4                  Bank           Gas Station    Athletics & Sports   

  6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
0               Dog Run      Doner Restaurant            Donut Shop   
1               Dog Run      Doner Restaurant            Donut Shop   
2   Rental Car Location        Medical Center                  Bank   
3   Distribution Center               Dog Run      Doner Restaurant   
4       Thai Restaurant                Bakery      Hakka Restaurant   

         9th Most Common Venue       10th Most Common Venue  
0          Dumpling Restaurant               Hardware Store  
1                    Drugstore  Eastern European Restaurant  
2               Discount Store          Distribution Center  
3                   Donut Shop                  Yoga Studio  
4  Eastern European Restaurant          Dumpling Restaurant

Now we can map the different clusters and see which neighborhoods are located near each other. Although a user might decide to move to a neighborhood for specific venues, they also might choose that location due to the neighborhoods nearby and what venues are located there. If neighborhoods nearby contain venues they would be interested in, it would not be far to travel to visit those places. This might make certain neighborhoods more attractive not just based the venues on the neighborhood itself, but also what venues are located in nearby neighborhoods.

In [31]:
map_clusters = folium.Map(location=[latitude, longitude], zoom_start=11)

x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

markers_colors = []
for lat, lon, poi, cluster in zip(toronto_merged['Latitude'], toronto_merged['Longitude'], toronto_merged['Neighborhood'], toronto_merged['ClusterLabels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color='red',
        fill=True,
        fill_color='yellow',
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

In [23]:
df1=toronto_merged.loc[toronto_merged['ClusterLabels'] ==1, toronto_merged.columns[[1] + list(range(5, toronto_merged.shape[1]))]]
df1

Community  ClusterLabels       1st Most Common Venue  \
0         Scarborough            1.0        Fast Food Restaurant   
1         Scarborough            1.0  Construction & Landscaping   
2         Scarborough            1.0                Intersection   
3         Scarborough            1.0                 Coffee Shop   
4         Scarborough            1.0        Caribbean Restaurant   
5         Scarborough            1.0                  Playground   
6         Scarborough            1.0               Train Station   
7         Scarborough            1.0                      Bakery   
8         Scarborough            1.0                       Motel   
9         Scarborough            1.0             College Stadium   
10        Scarborough            1.0           Indian Restaurant   
11        Scarborough            1.0           Accessories Store   
12        Scarborough            1.0                      Lounge   
13        Scarborough            1.0                 Pizza Place   
15        Scarborough            1.0        Fast Food Restaurant   
17         North York            1.0          Athletics & Sports   
18         North York            1.0              Clothing Store   
19         North York            1.0                        Café   
20         North York            1.0         Martial Arts School   
22         North York            1.0            Ramen Restaurant   
24         North York            1.0                 Pizza Place   
26         North York            1.0                         Gym   
27         North York            1.0                         Gym   
28         North York            1.0                        Bank   
29         North York            1.0        Caribbean Restaurant   
30         North York            1.0               Grocery Store   
31         North York            1.0               Grocery Store   
32         North York            1.0               Grocery Store   
33         North York            1.0               Grocery Store   
34         North York            1.0                 Pizza Place   
35          East York            1.0                 Pizza Place   
36          East York            1.0                Skating Rink   
37       East Toronto            1.0                       Trail   
38          East York            1.0                 Coffee Shop   
39          East York            1.0              Sandwich Place   
41       East Toronto            1.0            Greek Restaurant   
42       East Toronto            1.0                        Park   
43       East Toronto            1.0                 Coffee Shop   
45    Central Toronto            1.0            Department Store   
46    Central Toronto            1.0                 Coffee Shop   
47    Central Toronto            1.0                Dessert Shop   
48    Central Toronto            1.0                       Trail   
49    Central Toronto            1.0                 Coffee Shop   
51   Downtown Toronto            1.0                 Coffee Shop   
52   Downtown Toronto            1.0                 Coffee Shop   
53   Downtown Toronto            1.0                 Coffee Shop   
54   Downtown Toronto            1.0              Clothing Store   
55   Downtown Toronto            1.0                 Coffee Shop   
56   Downtown Toronto            1.0                 Coffee Shop   
57   Downtown Toronto            1.0                 Coffee Shop   
58   Downtown Toronto            1.0                 Coffee Shop   
59   Downtown Toronto            1.0                 Coffee Shop   
60   Downtown Toronto            1.0                 Coffee Shop   
61   Downtown Toronto            1.0                 Coffee Shop   
62         North York            1.0              Sandwich Place   
63    Central Toronto            1.0                 Music Venue   
65    Central Toronto            1.0                        Café   
66   Downtown Toronto            1.0                        Café   


In [24]:
clusters=pd.DataFrame({"Cluster1":df1["Community"],
                       
                      })

In [25]:
clusters = clusters.replace(np.nan, '', regex=True)

In [26]:
clusters

Cluster1
0         Scarborough
1         Scarborough
2         Scarborough
3         Scarborough
4         Scarborough
5         Scarborough
6         Scarborough
7         Scarborough
8         Scarborough
9         Scarborough
10        Scarborough
11        Scarborough
12        Scarborough
13        Scarborough
15        Scarborough
17         North York
18         North York
19         North York
20         North York
22         North York
24         North York
26         North York
27         North York
28         North York
29         North York
30         North York
31         North York
32         North York
33         North York
34         North York
35          East York
36          East York
37       East Toronto
38          East York
39          East York
41       East Toronto
42       East Toronto
43       East Toronto
45    Central Toronto
46    Central Toronto
47    Central Toronto
48    Central Toronto
49    Central Toronto
51   Downtown Toronto
52   Downtown Toronto
53   Downtown Toronto
54   Downtown Toronto
55   Downtown Toronto
56   Downtown Toronto
57   Downtown Toronto
58   Downtown Toronto
59   Downtown Toronto
60   Downtown Toronto
61   Downtown Toronto
62         North York
63    Central Toronto
65    Central Toronto
66   Downtown Toronto
67   Downtown Toronto
68   Downtown Toronto
69   Downtown Toronto
70   Downtown Toronto
71         North York
73               York
75   Downtown Toronto
76       West Toronto
77       West Toronto
78       West Toronto
80               York
81               York
82       West Toronto
83       West Toronto
84       West Toronto
85   Downtown Toronto
86        Mississauga
87       East Toronto
88          Etobicoke
89          Etobicoke
90          Etobicoke
91          Etobicoke
92          Etobicoke
94          Etobicoke
95          Etobicoke
96         North York
97         North York
99          Etobicoke
101         Etobicoke
102         Etobicoke

In [27]:
new_toronto=toronto_merged.set_index("Neighborhood",drop=True)
new_toronto.head(103)

Postal Code  \
Neighborhood                                                     
Malvern, Rouge                                             M1B   
Rouge Hill, Port Union, Highland Creek                     M1C   
Guildwood, Morningside, West Hill                          M1E   
Woburn                                                     M1G   
Cedarbrae                                                  M1H   
Scarborough Village                                        M1J   
Kennedy Park, Ionview, East Birchmount Park                M1K   
Golden Mile, Clairlea, Oakridge                            M1L   
Cliffside, Cliffcrest, Scarborough Village West            M1M   
Birch Cliff, Cliffside West                                M1N   
Dorset Park, Wexford Heights, Scarborough Town ...         M1P   
Wexford, Maryvale                                          M1R   
Agincourt                                                  M1S   
Clarks Corners, Tam O'Shanter, Sullivan                    M1T   
Milliken, Agincourt North, Steeles East, L'Amor...         M1V   
Steeles West, L'Amoreaux West                              M1W   
Hillcrest Village                                          M2H   
Fairview, Henry Farm, Oriole                               M2J   
Bayview Village                                            M2K   
York Mills, Silver Hills                                   M2L   
Willowdale, Willowdale East                                M2N   
York Mills West                                            M2P   
Willowdale, Willowdale West                                M2R   
Parkwoods                                                  M3A   
Don Mills                                                  M3B   
Don Mills                                                  M3C   
Bathurst Manor, Wilson Heights, Downsview North            M3H   
Northwood Park, York University                            M3J   
Downsview                                                  M3K   
Downsview                                                  M3L   
Downsview                                                  M3M   
Downsview                                                  M3N   
Victoria Village                                           M4A   
Parkview Hill, Woodbine Gardens                            M4B   
Woodbine Heights                                           M4C   
The Beaches                                                M4E   
Leaside                                                    M4G   
Thorncliffe Park                                           M4H   
East Toronto, Broadview North (Old East York)              M4J   
The Danforth West, Riverdale                               M4K   
India Bazaar, The Beaches West                             M4L   
Studio District                                            M4M   
Lawrence Park                                              M4N   
Davisville North                                           M4P   
North Toronto West, Lawrence Park                          M4R   
Davisville                                                 M4S   
Moore Park, Summerhill East                                M4T   
Summerhill West, Rathnelly, South Hill, Forest ...         M4V   
Rosedale                                                   M4W   
St. James Town, Cabbagetown                                M4X   
Church and Wellesley                                       M4Y   
Regent Park, Harbourfront                                  M5A   
Garden District, Ryerson                                   M5B   
St. James Town                                             M5C   
Berczy Park                                                M5E   
Central Bay Street                                         M5G   
Richmond, Adelaide, King                                   M5H   
Harbourfront East, Union Station, Toronto Islands          M5J   
Toronto Dominion Centre, Design Exchange                   M5K   
Commerce Court, Victoria 

### Compare Neighborhoods

Lets compare 2 neighborhoods within Toronto to see what are the most common venues in those areas. The neighborhoods we will test are Humber Summit in North York and Christie in Downtown Toronto. The 2 different neighborhoods we are testing are in different locations as a more northern area away that may be more family friendly, and another location that is located in the heart of the city which cater to different lifestyles in a busy environment.

In [28]:
Nei1=input("Enter the Neighborhood: ")

Enter the Neighborhood: Humber Summit


In [29]:
Nei2=input("Enter the Neighborhood: ")

Enter the Neighborhood: Christie


In [30]:
Venue_comparison=new_toronto.loc[[Nei1,Nei2]].T
Venue_comparison

Neighborhood                     Humber Summit            Christie
Postal Code                                M9L                 M6G
Community                           North York    Downtown Toronto
Latitude                               43.7563             43.6695
Longitude                              -79.566            -79.4226
ClusterLabels                                1                   1
1st Most Common Venue              Pizza Place       Grocery Store
2nd Most Common Venue   Furniture / Home Store                Café
3rd Most Common Venue                Drugstore                Park
4th Most Common Venue           Discount Store         Coffee Shop
5th Most Common Venue      Distribution Center  Italian Restaurant
6th Most Common Venue                  Dog Run         Candy Store
7th Most Common Venue         Doner Restaurant          Baby Store
8th Most Common Venue               Donut Shop  Athletics & Sports
9th Most Common Venue      Dumpling Restaurant           Nightclub
10th Most Common Venue      Dim Sum Restaurant          Restaurant

## Results and Discussion

There are many neighborhoods and even more venues to consider. If our clients or stakeholders had to do this by themselves manually, it would take hours if not days to even find a little information that there are looking for. Based on our analysis, we have found the 10 most common venues in each neighborhood. Our clients or end users may have the same top 3 venues there are looking for which would make several neighborhoods appear the same which would not provide enough distinction between them. With **2145** venues in **274** unique categories with **103** different communities with some of those communities having **2 or more** neighborhoods in each one. There is a lot of area to cover in order to pick the right neighborhood for our clients.

We started with listing all the neighborhoods with their geographical coordinates (longitude/latitude). After that we pulled FourSquare location data on all the different type venues located in the neighborhoods in Toronto. With all the location data on venues, we were able to organize them, and sort out the most common venues in each neighborhood. We cluster the neighborhoods together in their respective communities and print out a table. 

The most informative part of this application is the section that the end user or client would use most. After looking at the list of neighborhoods and their venues, they can compare neighborhoods or communities together side by side to give them a more visual comparison. We pulled **Humber Summit** and **Christie** neighborhoods together for a side by side test comparison on their 10 most common venue. As mentioned before, the top 3 most common venues may be the same for a number of people, but what this application has is the 10 most common venues which may provide a more specific picture on the make up of the neighborhoods which would narrow down the search for our clients or end users.

Based on the comparison we ran with our test neighborhoods, **Humber Summit** located in **North York** seems more appealing in our test based on the number of *restaurants* and *stores* that *families* might like rather than **Christie** located in **Downtown Toronto** that seems to provide a more city lifestyle that appeals more to *younger individuals or individuals without families* as it has venues like *nightclubs* and *athletics & sports*.

## Conclusion

The purpose of this application is to help our clients, end users, and stakeholders compare different neighborhoods in Toronto based on their most commone venues. As people may have their top 3 most common venues the same, what separates our application from others is that it list out the **10** most common venues which may help users separate the different neighborhoods that may have the same top 3 venues. The list also includes neighborhoods and venues in different parts of Toronto that may offer a vastly different lifestyle from each other that appeal to a number of different people.

Whether the user is single, young, and looking to live it up in the city or part of a family looking for a quiet family friendly part of town just outside the hustle and bustle of the big city, our application is here to help people distinguish the different communities and neighborhoods that may cater to each and everyone's lifestyle. In depth information provide will help separate the neighborhoods apart and even show neighborhoods nearby that our users may also be interested in if they can live in a neighborhood nearby and travel just a short distance to even more venues making their choice of neighborhood. Addition factors can be added such as housing prices, school rating, and more to this base application. Those factors may not matter to everyone as income and family size if different. The types of venus, location, and frequency of them is something that would matter to more people due to the number of different businesses that cater to people of all backgrounds. So no matter who you are, where you come from, this application can you help you find a neighborhood for you in Toronto!